Sources/Commands/SwiftTestTool.swift

```diff
@@ -209,3 +280,86 @@ private extension String {
         #endif
     }
 }
+
+/// A struct to hold the XCTestSuite data.
+struct TestSuite {
+
+    /// A struct to hold a XCTestCase data.
+    struct TestCase {
+        /// Name of the test case.
+        let name: String
+
+        /// Array of test methods in this test case.
+        let tests: [String]
+    }
+
+    /// The name of the test suite.
+    let name: String
+
+    /// Array of test cases in this test suite.
+    let tests: [TestCase]
+
+    /// Parses a JSON String to array of TestSuite.
+    ///
+    /// - Parameters:
+    ///     - jsonString: JSON string to be parsed.
+    ///
+    /// - Throws: JSONDecodingError, TestError
+    ///
+    /// - Returns: Array of TestSuite.
+    static func parse(jsonString: String) throws -> [TestSuite] {
+        let json = try JSON(bytes: ByteString(encodingAsUTF8: jsonString))
+        return try TestSuite.parse(json: json)
+    }
+
+    /// Parses the JSON object into array of TestSuite.
+    ///
+    /// - Parameters:
+    ///     - json: An object of JSON.
+    ///
+    /// - Throws: TestError
+    ///
+    /// - Returns: Array of TestSuite.
+    static func parse(json: JSON) throws -> [TestSuite] {
+        guard case let .dictionary(contents) = json,
+              case let .array(testSuites)? = contents["tests"] else {
+            throw TestError.invalidListTestJSONData
+        }
+
+        return try testSuites.map { testSuite in
+            guard case let .dictionary(testSuiteData) = testSuite,
+                  case let .string(name)? = testSuiteData["name"],
+                  case let .array(allTestsData)? = testSuiteData["tests"] else {
+                throw TestError.invalidListTestJSONData
+            }
+
+            let testCases: [TestSuite.TestCase] = try allTestsData.map { testCase in
+                guard case let .dictionary(testCaseData) = testCase,
+                      case let .string(name)? = testCaseData["name"],
+                      case let .array(tests)? = testCaseData["tests"] else {
+                    throw TestError.invalidListTestJSONData
+                }
+                let testMethods: [String] = try tests.map { test in
+                    guard case let .dictionary(testData) = test,
+                          case let .string(testMethod)? = testData["name"] else {
+                        throw TestError.invalidListTestJSONData
+                    }
+                    return testMethod
+                }
+                return TestSuite.TestCase(name: name, tests: testMethods)
+            }
+            return TestSuite(name: name, tests: testCases)
+        }
+    }
+
+    /// Method to list the tests inside testsuite on stdout.
+    /// Format: TestCaseName/TestName
+    func listTests() {
+        for testCase in tests {
+            for test in testCase.tests {
+                print(testCase.name + "/" + test)
+            }
+            print()
+        }
+    }
```

Comment:
I would prefer this just be inline in the definition of the `--list-tests` functionality. The reason is that this struct itself should probably eventually be factored into a lower-level module, but the high-level printing should stay here.